qiskit/cluster.py

```diff
@@ -73,7 +148,11 @@ def local_swap(self, qubit_index_1, qubit_index_2):
         self.qc.swap(self.main_reg[qubit_index_1], self.main_reg[qubit_index_2])
 
     def clear_ancillary(self):
-        self.qc.reset(self.teleport_reg)
+        try:
```

Comment:
I understand what you are doing here, but it may be clearer if you store the method in an attribute and make this conditional on method being set to teleport...

Reply:
This was a quick fix, I'll circle back to this